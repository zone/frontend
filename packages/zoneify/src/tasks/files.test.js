const { add, run } = require('./files')
const { writeFile } = require('../util/fileSystem')

jest.mock('../util/fileSystem')

writeFile.mockName('writeFile')

describe('tasks/files', () => {
  it('resolves if queue is empty', async () => {
    expect(await run()).toBeUndefined()
  })

  it('enqueues a file', async () => {
    writeFile.mockResolvedValue()

    add({ payload: { fileName: 'test1.md', content: 'test' } })
    add({
      payload: {
        fileName: 'test2.md',
        filePath: './test/test',
        content: 'test',
      },
    })

    const result = await run()

    expect(writeFile).toMatchSnapshot()
    expect(result).toBeInstanceOf(Array)
    expect(result.length).toBe(2)
  })

  it('handles thrown errors', async () => {
    writeFile.mockRejectedValue()

    add({ payload: { fileName: 'test1.md', content: 'test' } })

    await run().catch(error => expect(error).toMatchSnapshot())
  })
})
