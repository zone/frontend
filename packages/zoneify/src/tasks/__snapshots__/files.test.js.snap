// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tasks/files enqueues a file 1`] = `
[MockFunction writeFile] {
  "calls": Array [
    Array [
      "<PROJECT_ROOT>/test1.md",
      "test",
    ],
    Array [
      "<PROJECT_ROOT>/test/test/test2.md",
      "test",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`tasks/files handles thrown errors 1`] = `
[Error: Could not write [1mtest1.md[22m.
Check [1m<PROJECT_ROOT>/[22m exists and is writable.]
`;
